### ISSUE TITLE
Xamarin Forms App occasionally hangs during Resource.UpdateIdValues call

### ISSUE THREAD

**Issue by chrismcb** [xamarin/xamarin-android#1847]
This might be related to #1544 
About 1/4 the time the app boots, we hang during the Resources.UpdateIdValues call. Appears to happen in both debug and release builds.
I originally thought this was related to 1544 as we'll have about 50 calls to UpdateIdValues on the stack. But I added some code to UpdateIdValues to not allow it to be reentrant, and we still see the same hang.
If it hangs, it appears to hang on the line:
`global::Xamarin.Forms.Platform.Android.Resource.Attribute.actionBarSize = global::.....Droid.Resource.Attribute.actionBarSize;`
If I comment this line out, we won't hang.
The hang isn't consistent, but when it does happen, there is nothing on the stack. Almost like the stack was overwritten (again, making it seem like #1544 is the issue)
We have just under 4000 items in UpdateIdValues

Currently using VS 15.7.3 and this started happening when we switched to Xamarin Forms 3.0. We are considering switching back the previous release

**Comment by jonathanpeppers** [comment]
@chrismcb can you verify, is `UpdateIdValues` getting called 50 times?

Try this:
- Build your app, so `Resource.designer.cs` is generated
- Add `Android.Util.Log.Debug ("OOPS", "UpdateIdValues!")` to your app's `Resource.UpdateIdValues` method
- See how many logs you get

To confirm if this is a dup of #1544, you could add some code such as:
```
static bool updated = false;

public static void UpdateIdValues()
{
    if (updated)
        return;
    updated = true;
    //Other stuff that's already here
}
```

Unfortunately that code is going to get overwritten if you add/remove resources, update layouts, etc.

**Comment by chrismcb** [comment]
UpdatedIdValues is getting called 38 times. 
As I mentioned in my original comment, I already added code to make UpdateIdValues non reentrant (the "if updated" code you suggested I add) and this helped, but did not solve the problem.

**Comment by jonathanpeppers** [comment]
Ok, so what I'm getting at is that this is likely a duplicate of #1544 which is fixed in 103b5a755c048c6eaedb43c139cac32b920372cd. Don't use the code above, I think it will be best to get an updated Xamarin.Android.

The fix should be coming in future release of Visual Studio 15.8, I will ping you when it's available.

Closing for now, @chrismcb feel free to reopen, if you find something else here, thanks!

**Comment by chrismcb** [comment]
Thank you. I think we will go back to 2.5 for now, and I'll see if this is fixed when 15.8 comes out.
